File: streams-and-lambdas/src/main/java/de/rieckpil/learning/streamsandlambdas/Person.java
package de.rieckpil.learning.streamsandlambdas;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Person {

    private String name;
    private Gender gender;
    private int age;
}

File: kotlin-examples/src/main/kotlin/de/rieckpil/learning/either/Person.kt
package de.rieckpil.learning.either

data class Person(val name: Name, val age: Age)

data class Name(val value: String)

data class Age(val value: Int)

fun mkName(name: String): Either<String, Name> =
  if (name.isBlank()) Left("Name is empty.")
  else Right(Name(name))

fun mkAge(age: Int): Either<String, Age> =
  if (age < 0) Left("Age is out of range.")
  else Right(Age(age))

fun mkPerson(name: String, age: Int): Either<String, Person> =
  map2(mkName(name), mkAge(age)) { n, a -> Person(n, a) }
